.mailmap

```diff
@@ -14,163 +18,203 @@
-Dimitar Trenev <64981310+dtrenev@users.noreply.github.com>
 Dhruv Bhatnagar <61590679+dhruvbhq@users.noreply.github.com>
-Donny Greenberg <dongreenberg2@gmail.com>
-Donny Greenberg <dongreenberg2@gmail.com> <donny@ibm.com>
 Diego M. Rodríguez <diego.plan9@gmail.com>
-Drew Risinger <drewrisinger@users.noreply.github.com>
+Dimitar Trenev <64981310+dtrenev@users.noreply.github.com>
 Dmitri Maslov <34778124+dmitrimaslov@users.noreply.github.com>
+Douglas T. McClure <dtmcclur@us.ibm.com>
+Doğukan Tuna <5399244-sotapr@users.noreply.gitlab.com> <dtuname@gmail.com>
+Drew Risinger <drewrisinger@users.noreply.github.com>
+Edward H. Chen <1069980+ehchen@users.noreply.github.com>
+Edward H. Chen <1069980+ehchen@users.noreply.github.com> <ehchen@ibm.com>
 Edwin Navarro <enavarro@comcast.net>
+Elena Peña Tapia <57907331+ElePT@users.noreply.github.com>
+Elena Peña Tapia <57907331+ElePT@users.noreply.github.com> <epenatap@gmail.com>
+Eric Peterson <Eric.Peterson@ibm.com> <peterson.eric.c@gmail.com>
 Erick Winston <ewinston@us.ibm.com>
+Erick Winston <ewinston@us.ibm.com> <ewinston@users.noreply.github.com>
+Faisal Debouni <faisal@megoapps.com>
 Farai Mazhandu <35575882+faraimazh@users.noreply.github.com>
 Fran Cabrera <francabrera@users.noreply.github.com> <fcabrera@us.ibm.com>
-Fran Cabrera <francabrera@users.noreply.github.com> Fran Cabrera <francabrera@Frans-MacBook-Pro.local>
+Fran Cabrera <francabrera@users.noreply.github.com> <francabrera@Frans-MacBook-Pro.local>
 Franck Chevallier <fchevallier10@gmail.com>
+Frank Harkins <frankharkins@users.noreply.github.com> <frankharkins@hotmail.co.uk>
 Freya Shah <28627285+fs1132429@users.noreply.github.com>
-Gadi Aleksandrowicz <gadial@gmail.com>
+Freya Shah <28627285+fs1132429@users.noreply.github.com> <you@example.comfs1132429@gmail.com>
 Gabriele Agliardi <72732327+gabrieleagl@users.noreply.github.com>
-Gawel Kus <gawel.kus@ibm.com>
-Gawel Kus <gawel.kus@ibm.com> <gawel@oc7832844175.ibm.com>
+Gadi Aleksandrowicz <gadial@gmail.com>
+Gadi Aleksandrowicz <gadial@gmail.com> <gadia@il.ibm.com>
+George Zhou <geoyzhou@gmail.com>
 Georgios Tsilimigkounakis <45130028+georgios-ts@users.noreply.github.com>
-Hiroshi Horii <horii@jp.ibm.com> <HORII@jp.ibm.com>
+Georgios Tsilimigkounakis <45130028+georgios-ts@users.noreply.github.com> <ts.giorgos@yahoo.gr>
+Giacomo Nannicini <gnannicini@users.noreply.github.com> <giacomo.n@gmail.com>
+Gian Gentinetta <31244916+gentinettagian@users.noreply.github.com> <gian.gentinetta@gmx.ch>
+Ginés Carrascal <gines_carrascal@es.ibm.com> <gines.carrascal@gmail.com>
 Hiroshi Horii <horii@jp.ibm.com> <hhorii@users.noreply.github.com>
-Ian Gould <iangould@IANs-MacBook.local>
 Igor Olegovich Sokolov <iso@zurich.ibm.com> <ISO@zurich.ibm.com>
+Ikko Hamamura <ikkoham@users.noreply.github.com>
+Isha Rajput <33170219+irajput@users.noreply.github.com>
 Ismael Faro Sertage <ismael.faro1@ibm.com> <ismael.faro.sertage@gmail.com>
 Ismael Faro Sertage <ismael.faro1@ibm.com> <ismael.faro1@ibm.com>
```

Comment:
This rule might be matching on the name, but since the email's unchanged, it could probably be simplified to not specify the email twice.

Reply:
So, this is actually necessary, and the script tries to only generate such a line if its needed. git matches the record case-insensitively but replaces case sensitively and for things like shortlog treats emails that differ in case as different author records. So for the record (that actually exists in our history in commit 60b62d59595af22b7af045e815653e25c9623034 ):
`Ismael Faro <Ismael.Faro1@ibm.com>`
if we would have the mailmap line
`Ismael Faro Sertage <ismael.faro1@ibm.com>`
it would fixes only the name but
`Ismael Faro Sertage <ismael.faro1@ibm.com> <ismael.faro1@ibm.com>`
fixes the name and email.

Reply:
Huh, that's a fun corner case!